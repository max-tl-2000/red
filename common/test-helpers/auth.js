/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import moment from 'moment'; // eslint-disable-line red/no-moment
import { expect } from 'chai';

export const verifyTokenExpiration = (decodedToken, expectedTokenExpiration) => {
  const iat = moment.unix(decodedToken.iat);
  const exp = moment.unix(decodedToken.exp);

  const result = exp.diff(iat, expectedTokenExpiration.substring(expectedTokenExpiration.length - 1));
  const expectedValue = expectedTokenExpiration.substring(0, expectedTokenExpiration.length - 1);
  expect(result).to.equal(+expectedValue);
};
