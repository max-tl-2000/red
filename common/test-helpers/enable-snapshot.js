/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import chai from 'chai';
import chaiJestSnapshot from 'chai-jest-snapshot';

const enableSnapshot = () => {
  if (!global.before) return;
  chai.use(chaiJestSnapshot);
  before(() => {
    chaiJestSnapshot.resetSnapshotRegistry();
  });

  beforeEach(function beforeEach() {
    const { currentTest } = this;
    chaiJestSnapshot.setFilename(`${currentTest.file}.chai-snap`);
    chaiJestSnapshot.setTestName(currentTest.fullTitle());
  });
};

enableSnapshot();

export { editUnpredictable } from '../helpers/snapshot-helper';
