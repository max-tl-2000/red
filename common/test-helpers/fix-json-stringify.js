/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

const stringify = JSON.stringify;

before(() => {
  JSON.stringify = obj => {
    const seen = [];

    return stringify(obj, (key, val) => {
      if (typeof val === 'object') {
        if (seen.indexOf(val) >= 0) {
          return;
        }
        seen.push(val);
      }
      return val; // eslint-disable-line
    });
  };
});

after(() => {
  JSON.stringify = stringify;
});
