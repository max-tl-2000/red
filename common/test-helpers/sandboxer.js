/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

let _sinon;
let sandboxes = [];

function create() {
  // this is for when this module is used in karma
  _sinon = _sinon || global.sinon;
  const sandbox = _sinon.sandbox.create();
  sandboxes.push(sandbox);

  return sandbox;
}

function restore() {
  sandboxes.forEach(sandbox => sandbox.restore());
  sandboxes = [];
}

function setSinon(sinon) {
  _sinon = sinon;
}

export { create, restore, setSinon };
