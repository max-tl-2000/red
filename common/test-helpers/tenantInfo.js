/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

export const TEST_TENANT_ID = 'af20764d-1e0d-41a5-be54-3766215f89a4';
