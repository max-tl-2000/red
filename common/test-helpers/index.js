/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import chai from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';
import { unmountComponentAtNode } from 'react-dom';
import when from './when';
import and from './and';
import * as overrider from './overrider';
import * as sandboxer from './sandboxer';
import sleep from '../helpers/sleep';

sandboxer.setSinon(sinon);

const expect = chai.expect;
const assert = chai.assert;

chai.use(sinonChai);

let div;

if (global.document) {
  beforeEach(() => {
    div = document.createElement('div');
    div.setAttribute('id', 'sandbox');

    document.body.appendChild(div);
  });
}

afterEach(() => {
  if (global.document && div) {
    unmountComponentAtNode(div);
    document.body.removeChild(div);
    div = null;
  }

  overrider.restore();
  sandboxer.restore();
});

const getSandBoxDiv = () => div;

// helper method to trigger a real click event in the browser
// jquery.fn.click is not the same as this as the event need
// to also generate the capture phase
const triggerMouseEvent = (ele, evt) => {
  const mEvent = document.createEvent('MouseEvent');
  // interface
  // event.initMouseEvent(type, canBubble, cancelable, view,
  //                    detail, screenX, screenY, clientX, clientY,
  //                    ctrlKey, altKey, shiftKey, metaKey,
  //                    button, relatedTarget);
  mEvent.initMouseEvent(evt, true, true, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null);

  ele.dispatchEvent(mEvent);
};

export { getSandBoxDiv, triggerMouseEvent, overrider, expect, assert, sinon, when, sandboxer, and, sleep };
