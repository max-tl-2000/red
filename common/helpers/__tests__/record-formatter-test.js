/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import { readFileSync } from 'fs'; // had to use readFileSync, for some reason xfs.read was not working
import path from 'path';
import expand from 'glob-expand';
import { when } from '../../test-helpers';
import recordFormatter from '../record-formatter';
import tryParse from '../try-parse';

/**
 * reads the sample json files from the provided glob
 *
 * @return     {Array}          array of objects with the following structure:
 * {
 *   name,          the name of the record key. This is inferred from the first
 *                  part of the filename before the -- mark
 *   expectedFile,  the name of the expected file to compare the result against
 *   rec,           the parsed JSON entity to be passed to objectFormatter
 * }
 */
const readSampleRecords = () => {
  const pattern = path.join(__dirname, './fixtures/recs/**/*.json');
  const recordFiles = expand(pattern);
  return recordFiles.map(f => ({
    name: path.basename(f, '.json'),
    rec: tryParse(readFileSync(f, { encoding: 'utf8' }), {}),
  }));
};

describe('recordFormatter.format', () => {
  when('format receives an object like:', () => {
    const recordExamples = readSampleRecords();

    recordExamples.forEach(entry => {
      it(`should format ${entry.name} as expected`, async () => {
        // the formatter return an array that is passed to console.log
        // console.log will just concatenate it and send it to stdout

        const { rec } = entry;
        // this is needed since the Buffer type is not serializable
        if (rec.subType === 'AMQP' && rec.amqpMessage.content && rec.amqpMessage.content.type === 'Buffer') {
          rec.amqpMessage.content = Buffer.from(rec.amqpMessage.content.data);
        }
        const formatted = recordFormatter.format(rec);
        const result = formatted.join(' ');
        expect(result).toMatchSnapshot(entry.name);
      });
    });
  });
});
