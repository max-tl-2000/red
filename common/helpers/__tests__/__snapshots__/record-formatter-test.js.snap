// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recordFormatter.format when format receives an object like: should format access-log-req as expected: access-log-req 1`] = `
"[46m[30m[server]:[39m[49m 
[36mâ‡ â‡  [eb246369-4996-4791-84c8-c05c2aed9282] â‡ â‡ 
[39m[90m  GET /scripts-dev/__webpack_hmr
[39m[90m  headers: 
[39m[90m    host: red.localhost.com:3000
    connection: keep-alive
    pragma: no-cache
    cache-control: no-cache
    accept: text/event-stream
    user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.106 Safari/537.36
    referer: http://red.localhost.com:3000/
    accept-encoding: gzip, deflate, sdch
    accept-language: en-US,en;q=0.8,es;q=0.6
    cookie: i18next=en-US
[39m[36mâ‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ 
[39m
"
`;

exports[`recordFormatter.format when format receives an object like: should format access-log-res as expected: access-log-res 1`] = `
"[46m[30m[server]:[39m[49m 
[33mâ‡¢â‡¢ [4dfcb8cd-4c30-46eb-bf1d-eca656a41982] â‡¢â‡¢
[39m[90m  status: [32m200[90m
[39m[90m  duration (ms): 10
[39m[90m  headers: 
[39m[90m    HTTP/1.1 200 OK
    X-Powered-By: Express
    X-Request-Id: 4dfcb8cd-4c30-46eb-bf1d-eca656a41982
    Set-Cookie: i18next=en-US; path=/; expires=Sun, 02 Jul 2017 03:15:14 GMT
    Content-Type: text/html; charset=utf-8
    ETag: W/\\"9db-cLcxPlPR976GFlmuXCJTvQ\\"
    Vary: Accept-Encoding
    Content-Encoding: gzip
    Date: Sat, 02 Jul 2016 03:15:14 GMT
    Connection: keep-alive
    Transfer-Encoding: chunked
[39m[33mâ‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢
[39m
"
`;

exports[`recordFormatter.format when format receives an object like: should format access-log-unknown as expected: access-log-unknown 1`] = `"[46m[30m[server]:[39m[49m unknown message in the access-log"`;

exports[`recordFormatter.format when format receives an object like: should format amqp-publish as expected: amqp-publish 1`] = `
"[46m[30m[integration-test]:[39m[49m [37m[t]: [AMQP] publishing to integration_app_exchange tenant_created
[39m[90mamqpMessage: {
  id: 'f342fdbe-4e6e-485e-8243-bbd0cb2864ec',
  name: 'red',
  authorization_token: 'edadbb9e-021b-4a8f-af95-37fe43fe396d'
}[39m"
`;

exports[`recordFormatter.format when format receives an object like: should format amqp-received as expected: amqp-received 1`] = `
"[46m[30m[integration-test]:[39m[49m [37m[t]: [AMQP] Received in consumer=0 from exchange: integration_app_exchange and topic: new_user_registered
[39m[90mamqpMessage: {
  content: '{\\"name\\":\\"Jon\\",\\"tenantId\\":\\"6ebd4b29-b6b4-4963-81e9-baaad9bb3c53\\",\\"userId\\":\\"48e0c910-2cfe-4660-907e-f2b871786fbe\\"}',
  fields: {
    consumerTag: 'amq.ctag-x-Yh19HWfz_SREl0-34Vhg',
    deliveryTag: 1,
    redelivered: false,
    exchange: 'integration_app_exchange',
    routingKey: 'new_user_registered'
  },
  properties: { headers: {}, deliveryMode: 2 }
}[39m"
`;

exports[`recordFormatter.format when format receives an object like: should format default as expected: default 1`] = `
"[46m[30m[integration-test]:[39m[49m [1m[34m08:53:32.379[39m[22m [33m[d]:[39m save tenant result
 subType: 'tenantService'
 result: {
  created_at: null,
  updated_at: null,
  id: 'b075d0a6-d1ab-4477-9e5b-0a7961d1c5df',
  name: 'red',
  migrations_path: null,
  phoneNumber: null,
  authorization_token: '80907417-c664-4c10-b906-3c9fba0a62d8',
  metadata: null,
  refreshed_at: '2016-06-30T08:53:31.609Z'
}
"
`;

exports[`recordFormatter.format when format receives an object like: should format i18n as expected: i18n 1`] = `"[46m[30m[server]:[39m[49m [37m[t]: i18n - currentLng set to: en-US[39m"`;

exports[`recordFormatter.format when format receives an object like: should format i18n--error as expected: i18n--error 1`] = `
"[46m[30m[server]:[39m[49m [1m[34m03:15:14.420[39m[22m [31m[e]:[39m currentLng set to: en-US
 subType: 'i18n'
 args: []
 [31mReferenceError: color is not defined
    at getColorForRecordBasedOnLevel (record-formatter.js:22:5)
    at Object.format (record-formatter.js:88:42)
    at Logger.overrideLevel [as _emit] (logger.js:53:41)
    at Server.<anonymous> (server.js:142:12)
    at Server.g (events.js:273:16)
    at emitNone (events.js:85:20)
    at Server.emit (events.js:179:7)
    at emitListeningNT (net.js:1253:10)
    at nextTickCallbackWith1Arg (node.js:464:9)[39m

 [90m  /Users/roy/royws/red/server/helpers/i18n.js:34[39m
"
`;

exports[`recordFormatter.format when format receives an object like: should format uncaught-exception as expected: uncaught-exception 1`] = `
"[46m[30m[server]:[39m[49m [31m[Unhandled exception][39m
  [90mpid: 45712
  [31mundefined [NO STACK PROVIDED][90m
[39m
"
`;

exports[`recordFormatter.format when format receives an object like: should format unhandled-rejection as expected: unhandled-rejection 1`] = `
"[46m[30m[integration-test]:[39m[49m [31m[Unhandled rejection][39m
  [90mpid: 56627
  [31mReferenceError: color is not defined
    at getColorForRecordBasedOnLevel (record-formatter.js:22:5)
    at Object.format (record-formatter.js:88:42)
    at Logger.overrideLevel [as _emit] (logger.js:53:41)
    at Server.<anonymous> (server.js:142:12)
    at Server.g (events.js:273:16)
    at emitNone (events.js:85:20)
    at Server.emit (events.js:179:7)
    at emitListeningNT (net.js:1253:10)
    at nextTickCallbackWith1Arg (node.js:464:9)[90m
[39m
"
`;
